### ISSUE TITLE
model attributes of `WIMPEnergySpectrum` not loaded in correctly using configuration files

### ISSUE THREAD

**Issue by plt109** [FlamTeam/flamedisx#210]
Context: I was trying to simulate some `fd.SR1WIMPSource` events while loading the model parameters in via the configuration file functionality ([PR 140](https://github.com/FlamTeam/flamedisx/pull/140)).
```config_path = '/home/peaelle42/analyses/field_inhomogenieties/SR1WIMPSource_config_diff.py'
df = fd.SR1WIMPSource(config=config_path).simulate(100)
```
Specifically, I wanted to simulate some `fd.SR1WIMPSource` events between 30 keV - 40 keV instead of the default 0.7 - 50 keV so I had the following line in the config file
```
energy_edges                            = np.geomspace(30., 40, 100)
```
However, I got some simulated events with energy outside 30 keV - 40 keV.

Problem: The model attributes of `WIMPEnergySpectrum` were not loaded in correctly from configuration files.

The configurations are loaded in in [`source.py`](https://github.com/FlamTeam/flamedisx/blob/a76b2d58b1fbd38bcd0ce28a0796fd12e61b9861/flamedisx/source.py#L301) which is called after the energy spectrum has been computed in [`energy_spectrum.py`](https://github.com/FlamTeam/flamedisx/blob/2ed08d06c4aca38f6fd7e0e60c29eca9aa3ffeb3/flamedisx/block_source.py#L252).

Solution:
It is possible to create a specific class that inherits from `fd.SR1WIMPSource` and this somehow manages to overwrite the model attributes like `mw` and `energy_edges`.

```
class my_wimp(fd.SR1WIMPSource):
    mw = 50
    energy_edges = np.geomspace(30., 40., 100)
```

However, being able to load in all model attributes of the sources via a single configuration file is still a better solution in terms of keeping track of the configurations used for fits.

**Comment by JelleAalbers** [comment]
Good catch! The problem is that the WIMPsource computes and sets new attributes (energies and energy_hist) on initialization/setup, and does not recompute them when the original attributes (energy_edges, mw, etc.) are changed. 

I think the fix would be to have BlockModelSource override set_defaults to call each Block.setup() again after Source.set_defaults. The one complication would be that the length of array columns like energy_edges might change as well, so we'd either have to disallow that (so you probably still have to do subclassing) or make the handling of the array_columns field more complex.

I was working on a fix when I realized the NEST sources currently use copy-paste-modified energy spectra -- so any changes would have to be there as well, and there is a big PR for that open already. If this is not a big roadblock, I would wait until that is merged and we've had a change to reduce some of the duplication.
